### PR TITLE
Add pause/unpause/inspect integration test

### PR DESCRIPTION
**Pull request by jimmyxian**
Add pause/unpause/inspect integration test
Signed-off-by: Xian Chaobo <xianchaobo@huawei.com>

### CODE REVIEW

**Comment by abronan** [comment]
Weirdly, `pause` cannot complete in my setup. It makes the `dind` container go into the `<defunct>` state and the test stays blocked forever (probably on `docker unpause`).

/cc @aluzzardi @vieux

**Comment by abronan** [comment]
Does it work for you @jimmyxian?

**Comment by aluzzardi** [comment]
LGTM once @abronan issue is solved

**Comment by aluzzardi** [comment]
Related to #492

File: test/integration/api.bats [hunk 1]
```diff
@@ -86,9 +86,41 @@ function teardown() {
 	[[ "${lines[3]}" == *"Nodes: 3" ]]
 }
 
-# FIXME
 @test "docker inspect" {
-	skip
+	start_docker 3
```
Comment (aluzzardi) [inline test/integration/api.bats]: @vieux: Should we check the `Node` field that we add to `docker inspect` here?

File: test/integration/api.bats [hunk 2]
```diff
@@ -121,9 +153,27 @@ function teardown() {
 	skip
 }
 
-# FIXME
 @test "docker pause" {
-	skip
+	start_docker 3
```
Comment (aluzzardi) [inline test/integration/api.bats]: LGTM

Please check if it works on your machine as @abronan suggested.

File: test/integration/api.bats [hunk 3]
```diff
@@ -232,9 +282,36 @@ function teardown() {
 	skip
 }
 
-# FIXME
 @test "docker unpause" {
-	skip
+	start_docker 3
```
Comment (aluzzardi) [inline test/integration/api.bats]: LGTM

**Comment by jimmyxian** [comment]
@abronan @aluzzardi @vieux 
Yeah, It's a bug. If the state of the container is paused, it can't be removed(`rm -f`), so it block in stop_docker. So three ways to solve:
+ upause it after paused sucessfully
+ modify stop_docker in helpers.bash: unpause all the paused container before `rm -f`
+ fix it in docker: `rm -f` should remove container in any states

**Comment by abronan** [comment]
@jimmyxian Imho for now the better solution would be to `pause` and then `unpause` the container for `stop_docker` to complete inside the `docker pause` test. As we are using `pause` only in this test, we don't want to have an extra check for paused containers after each set of tests.

Regarding the fix in docker, it's a well known issue that ghosted containers cannot be cleaned up properly. So just adding the `unpause` to the test will do for now.

Just tested and it works fine, If you could add this to the existing test that would be great! Thanks!

**Comment by aluzzardi** [comment]
Agree with @abronan 

Also, maybe we should add a check of `Node` in inspect - I'll let @vieux decide.

File: test/integration/api.bats (hunk 1 above)
Comment (vieux) [inline test/integration/api.bats]: yes very good idea!

ew should look for a `  "Node": {` object and a maybe `"Name": "node-"` as all the node start with `node-`

**Comment by jimmyxian** [comment]
@abronan @aluzzardi @vieux  thks
Add `unpause` in pause test and check node infomation in `inspect`

**Comment by abronan** [comment]
@jimmyxian Perfect. Thanks!

LGTM

**Pull request merged**